pkg/controllers/resourceapply/config_hash.go

```diff
@@ -0,0 +1,136 @@
+package resourceapply
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const configsHashAnnotation = "operator.openshift.io/configs-hash"
```

Comment:
I would make this `configHash` rather than `configsHash`
```suggestion
const configHashAnnotation = "operator.openshift.io/config-hash"
```